src/backend/distributed/commands/index.c

```diff
@@ -168,6 +168,67 @@ PlanIndexStmt(IndexStmt *createIndexStatement, const char *createIndexCommand)
 }
 
 
+/*
+ * PlanReindexStmt determines whether a given REINDEX statement involves
+ * a distributed table. If so, error until support is implemented.
+ */
+List *
+PlanReindexStmt(ReindexStmt *ReindexStatement, const char *ReindexCommand)
+{
+	List *ddlJobs = NIL;
+
+	/*
+	 * We first check whether a distributed relation is affected. For that, we need to
+	 * open the relation. To prevent race conditions with later lookups, lock the table.
+	 */
+	if (ReindexStatement->relation != NULL)
```

Comment:
Minor suggestion in order to decrease the indentation level:

```
if (ReindexStatement->relation == NULL)
{

   /* we don't do anything for REINDEX_OBJECT_SCHEMA/ REINDEX_OBJECT_SYSTEM/REINDEX_OBJECT_DATABASE at the moment */
     return;
}
```

In the future, we might consider erroring out if the schema/database/system has any distributed tables

Reply:
We definitely should not error out if there are distributed tables. It'd only make sense if there were only distributed tables

Reply:
To me, it makes more sense to error out (or at least WARNING) with a clear message. Because the user expects all the tables in the schema/db/system to be reindexed once the command succeeds. 

The user could reindex the local tables one by one if needed. But we can discuss that when we need it :)